tendermint/src/public_key.rs

```diff
@@ -49,6 +53,7 @@ impl PublicKey {
 
     /// Get Ed25519 public key
     pub fn ed25519(self) -> Option<ed25519::PublicKey> {
+        #[allow(unreachable_patterns)]
```

Comment:
I understand that those `_ => unreachable!()` patterns were here previously to this PR, but just wanted to mention that those seem inherently unsafe to me (as in: may lead to panics). This PR improves the state of things by making them actually unreachable if the `secp265k1` feature is not enabled, but we should perhaps take a closer look at them in a follow-up issue.